### ISSUE TITLE
stack-buffer-overflow at DecodeGifImg() in ngiflib.c:554

### ISSUE THREAD

**Issue by Marsman1996** [miniupnp/ngiflib#10]
Tested in Ubuntu 16.04, 64bit, ngiflib(master c8488d5) 

Triggered by
`$ gif2tga $POC`

POC file:
https://github.com/Marsman1996/pocs/blob/master/ngiflib/poc12-DecodeGifImg-stackoverflow

ASAN info:
```
==122055==ERROR: AddressSanitizer: stack-buffer-overflow on address 0x7ffddb48cc0f at pc 0x56351fd192cd bp 0x7ffddb4899e0 sp 0x7ffddb4899d0
WRITE of size 1 at 0x7ffddb48cc0f thread T0
    #0 0x56351fd192cc in DecodeGifImg /home/ubuntu/Desktop/crashana/ngiflib/ngiflib-c8488d5/ngiflib.c:554
    #1 0x56351fd1ad51 in LoadGif /home/ubuntu/Desktop/crashana/ngiflib/ngiflib-c8488d5/ngiflib.c:802
    #2 0x56351fd15d84 in main /home/ubuntu/Desktop/crashana/ngiflib/ngiflib-c8488d5/gif2tga.c:95
    #3 0x7f14eb381b96 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21b96)
    #4 0x56351fd15129 in _start (/home/ubuntu/Desktop/crashana/ngiflib/ngiflib-c8488d5/gif2tga+0x2129)

Address 0x7ffddb48cc0f is located in stack of thread T0 at offset 12703 in frame
    #0 0x56351fd1756d in DecodeGifImg /home/ubuntu/Desktop/crashana/ngiflib/ngiflib-c8488d5/ngiflib.c:387

  This frame has 4 object(s):
    [32, 320) 'context'
    [352, 8544) 'ab_prfx'
    [8576, 12672) 'ab_suffx'
    [12704, 16800) 'ab_stack' <== Memory access at offset 12703 underflows this variable
HINT: this may be a false positive if your program uses some custom stack unwind mechanism or swapcontext
      (longjmp and C++ exceptions *are* supported)
SUMMARY: AddressSanitizer: stack-buffer-overflow /home/ubuntu/Desktop/crashana/ngiflib/ngiflib-c8488d5/ngiflib.c:554 in DecodeGifImg
Shadow bytes around the buggy address:
  0x10003b689930: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x10003b689940: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x10003b689950: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x10003b689960: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x10003b689970: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 f2 f2
=>0x10003b689980: f2[f2]00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x10003b689990: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x10003b6899a0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x10003b6899b0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x10003b6899c0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x10003b6899d0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07
  Heap left redzone:       fa
  Freed heap region:       fd
  Stack left redzone:      f1
  Stack mid redzone:       f2
  Stack right redzone:     f3
  Stack after return:      f5
  Stack use after scope:   f8
  Global redzone:          f9
  Global init order:       f6
  Poisoned by user:        f7
  Container overflow:      fc
  Array cookie:            ac
  Intra object redzone:    bb
  ASan internal:           fe
  Left alloca redzone:     ca
  Right alloca redzone:    cb
==122055==ABORTING
```

**Comment by miniupnp** [comment]
I do not reproduce. (I've built with `-fsanitize=address`)
Can you be more specific about your build options ?
also please check with latest master (3bb9980edf43087335058e2a3a66a25eed185859)
I believe it may already be fixed in f1b3bb80fb66b0d92876b61bd9a466c59dbbe652

**Comment by miniupnp** [comment]
OK I checked. The bug is in c8488d5, but is fixed in latest master. Please update your sources to the latest master before reporting a bug.
duplicate of #9
